p2p/dht/findnode.go

```diff
@@ -24,6 +23,8 @@ const protocol = "/dht/1.0/find-node/"
 // ErrEncodeFailed is returned when we failed to encode data to byte array
 var ErrEncodeFailed = errors.New("failed to encode data")
 
+var swarmconfig  config.SwarmConfig
```

Comment:
This shouldn't be a "static" variable but rather a part of the struct, the config struct should be passed to the FindNodeProtocol constructing method. this way if we have multiple FindNode instances running in the same binary their configs won't mess with each other. its also easier to test like that.